### ISSUE TITLE
Too broad keycode -> space cast

### ISSUE THREAD

**Issue by QRiary** [Ikcelaks/qmk_sequence_transform#43]
sequence_transform.c -> process_sequence_transform -> line 489

default case there casts a lot of stuff it shouldn't to space, which causes enhanced backspace go wild, should add cases for arrows, enter and such to actually reset the buffer

**Comment by Ikcelaks** [comment]
I think that we should have the default here be to reset the buffer and return false, if the key isn't a letter, number, symbol, or space.

IMO, even the other white space characters, like tab and enter should reset the buffer. This still accomplishes the most useful thing of setting a space at the start of the buffer.

Thoughts @kamih ?

**Comment by kamih** [comment]
Ok. This and the issue #37 are the same.
We need to clearly define default rules (in code) for which keycodes get translated to `KC_SPACE`, and which keycodes trigger a bluffer clear (with `KC_SPACE` push). Then the script should output something that allows the code to override this behavior on a key per key basis. For example if a rule uses a certain symbol, it should not clear the buffer.
And then on top of that, do we want to let the user override this again?
So basically, we need a function that gets passed a pointer to the key. It can change it to space if needed, and it can clear the buffer if needed.
I think.

**Comment by QRiary** [comment]
No, for #37 I like the suggested idea of having [such](https://github.com/QKekos/qmk_firmware/blob/master/keyboards/beekeeb/piantor/keymaps/qkekos/magic_sturdy/general.c#L42-L67) cast function, I probably should close that issue, but that's user-api, but this issue is about our code, its not just some user decisions whether to cast key `a` to key `b` or not, that's our logic HOW to handle the key we get, like casting to space was fine with no enhanced backspace, but now it should become stricter

**Comment by Ikcelaks** [comment]
Metacharacters greatly enhanced this behavior. This issue is no longer relevant.
